### PR TITLE
Avoided redundant handleChange call

### PR DESCRIPTION
**Pull request by Amaljith-K**
**Description**
Previously onBlur was causing a redundant onChange call. Now onChange will be called only if the value contains leading or trailing whitespaces.

**Checklist**

- [x] ~~I have made corresponding changes to the documentation.~~
- [x] ~~I have updated the types definition of modified exports.~~
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**
@ajmaln _a
cc @navaneethsdk thanks for pointing out this bug in https://github.com/bigbinary/smile-cart-frontend/pull/57

### CODE REVIEW

File: src/components/Input.jsx [hunk 1]
```diff
@@ -68,7 +68,10 @@ const Input = forwardRef(
 
     const handleOnBlur = e => {
       if (!disableTrimOnBlur && typeof value === "string") {
-        e.target.value = value.trim();
+        const trimmedValue = value.trim();
+        if (value === trimmedValue) return;
```
Comment (ajmaln) [inline src/components/Input.jsx:72]: We can't return here, it would prevent the custom `onBlur` handler from triggering. We can conditionally call `handleChange` instead.

**Review by ajmaln — COMMENTED** [review]
@Amaljith-K _a Please check

**Comment by Amaljith-K** [comment]
@ajmaln _a thanks for pointing this out. It has been fixed and tests have been added to cover that edge case. Please review.

**Review by ajmaln — APPROVED** [review]
